content/200-concepts/100-components/01-prisma-schema/07-referential-actions.mdx

```diff
@@ -0,0 +1,329 @@
+
+2. Upgrading Prisma to 2.26.0 and _**enabling**_ the feature flag **without** adding custom referential actions. (Using the **new** defaults)
+
+  If you want to keep your current database constraints but upgrade you can enable the feature flag. To do so you must have a required relation in your schema, then run either a Prisma Migration or `db push`, this will change the default for `ON DELETE` from `CASCADE` TO `RESTRICT`.
+
+  The change is relevant for both the Prisma Client _and_ Migration. If an attempt is made to delete a referenced object, if their are related records, a database error will be thrown.
+
+3. Upgrading Prisma to 2.26.0, _**enabling**_ the feature flag and **adding custom referential actions**.
+
+  Once you have upgraded the Prisma version and added the feature flag to the generator block you can begin using the referential actions.
+
+  ### Using Introspection
+
+  You can begin by running an [Introspection](../introspection) on your database. This will update your data modals to use `ON DELETE CASCADE` and explicitly set `onDelete: Cascade` on your modals `@relation` fields. 
+
+  The happens because the the old defaults (`ON DELETE CASCADE`) are different from the new (`ON DELETE RESTRICT`), so when the introspection is run, it will see the old default and use that to explicitly set the referential action on the relation.
+
+  Once the introspection has run and the referential action has been set, you can change the `Post` author relation to `onDelete: SetNull`.
+
+  ### Using Migration
+
+  You can begin by running a [Migration](../prisma-migrate) (or `db push`) on your database. This will change the default referential actions from `ON DELETE CASCADE` to `ON DELETE RESTRICT`. 
+  
+  Next you can manually add a custom referential action to the `User` relation on the `Profile` modal by adding the following line after the `references` property; `onDelete: Cascade`. 
+  
+  Finally you can change the `Post` author relation to `onDelete: SetNull`.
+
+  ```prisma file=schema.prisma
+  model Profile {
+    user   User    @relation(fields: [userId], references: [id], onDelete: Cascade)
+    userId Int     @unique
+  }
+
+  model Post {
+    author    User?     @relation(fields: [authorId], references: [id], onDelete: SetNull)
+    authorId  Int?
+  }
+  ```
+
+  If you then choose to delete a `User`, their corresponding `Profile` records will be deleted and the `authorId` scalar field on the `Post` table will be set to null.
+
+## Important caveats
+
+If you are using the [Introspection](../introspection) workflow it's important to be aware of how the changes to the default referential actions and their subsequent behaviors could potentially affect your data. 
+
+When you run an Introspection the [old](#current-behavior) default referential actions in the database will be reflected in the Prisma schema (See [Using Introspection](#using-introspection)). 
```

Comment:
So, meaning, your required relations will have `onDelete: Cascade` after an introspection, and if you continue using the data model, the cascading will work. If you instead migrate without setting `onDelete` in any of the models, the database foreign key actions will be changed to `RESTRICT` and the old cascading behavior with `raw` will not work anymore.

Reply:
Oh i think ive got it now! Haha i just posted in slack to ask to make this clearer but i think i get you